src/common/config_proxy.h

```diff
@@ -81,11 +81,16 @@ class ConfigProxy {
 
   std::map<md_config_obs_t*, CallGateRef> obs_call_gate;
 
-  void call_observers(rev_obs_map_t &rev_obs) {
+  void call_observers(std::unique_lock<ceph::recursive_mutex>& locker,
+                      rev_obs_map_t& rev_obs) {
+    // observers are notified outside of lock
+    locker.unlock();
     for (auto& [obs, keys] : rev_obs) {
       obs->handle_conf_change(*this, keys);
-      // this can be done outside the lock as call_gate_enter()
-      // and remove_observer() are serialized via lock
+    }
+    locker.lock();
+
+    for (auto& [obs, keys] : rev_obs) {
```

Comment:
unused variable `obs` warning:

https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=leap15,DIST=leap15,MACHINE_SIZE=huge/29459//consoleFull

(It's really annoying we can't disable that somehow : / Structured binding is really useful for these loops...)

Reply:
Tweaked